packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -332,6 +360,14 @@ export class ContextsManager {
       }
     }
 
+    // Restart informers for secondary resources if watchers are subscribing for this resource
+    for (const resourceName of secondaryResources) {
+      if (this.secondaryWatchers.hasSubscribers(resourceName)) {
```

Comment:
Needs a check like in getCurrentContextResources():
`if (this.secondaryWatchers.hasSubscribers(resourceName) && !this.states.hasInformer(this.currentContext, resourceName)) {`
This fixes the regression I saw while testing.

Reply:
no, moving into this block instead:
```
    if (contextChanged) {

```